File: impl/ts/src/markdown_tests.ts
import { readFileSync } from "node:fs";
import { canonicalizeMarkdownProfile, validateMarkdownProfile } from "./markdown.js";

const path = process.argv[2] ?? "spec/agentnet-markdown-tests-v0.1.json";
const data = JSON.parse(readFileSync(path, "utf-8"));

for (const entry of data.cases as any[]) {
  const id = entry.id as string;
  const input = entry.input as string;
  const canonical = entry.canonical as string;
  const valid = entry.valid as boolean;

  try {
    const normalized = canonicalizeMarkdownProfile(input);
    if (normalized !== canonical) {
      throw new Error(`${id} canonical mismatch`);
    }
  } catch (err) {
    if (canonical !== "") {
      throw err;
    }
  }

  let isValid = true;
  try {
    validateMarkdownProfile(input);
  } catch {
    isValid = false;
  }
  if (isValid !== valid) {
    throw new Error(`${id} validity mismatch`);
  }
}

console.log("markdown profile tests complete");

File: impl/python/agentnet_py/markdown_tests.py
import json
import sys
from pathlib import Path

from .markdown import canonicalize_markdown_profile, validate_markdown_profile


def main():
    path = Path(sys.argv[1]) if len(sys.argv) > 1 else Path("spec/agentnet-markdown-tests-v0.1.json")
    data = json.loads(path.read_text())
    for case in data["cases"]:
        case_id = case["id"]
        input_text = case["input"]
        canonical = case["canonical"]
        valid = case["valid"]
        try:
            normalized = canonicalize_markdown_profile(input_text)
        except Exception:
            if canonical != "":
                raise AssertionError(f"{case_id} canonicalization failed")
        else:
            if normalized != canonical:
                raise AssertionError(f"{case_id} canonical mismatch")
        is_valid = True
        try:
            validate_markdown_profile(input_text)
        except Exception:
            is_valid = False
        if is_valid != valid:
            raise AssertionError(f"{case_id} validity mismatch")
    print("markdown profile tests complete")


if __name__ == "__main__":
    main()
